lib/lua/lua_init.lua

```diff
@@ -65,3 +65,59 @@ config.register_key = function(key)
         error("keybinding action: expected string or a function"..use_key, 2)
     end
 end
+-- Register callback to execute on restart
+way_cooler_table.on_restart = function(callback)
+    assert(callback, "missing callback")
+    assert(type(callback) == 'function', "callback: expected function")
+    __rust.restart_callback = callback
+end
+-- Register a function to execute on terminate
+way_cooler_table.on_terminate = function(callback)
+    assert(callback, "missing callback")
+    assert(type(callback) == 'function', "callback: expected function")
+    __rust.on_terminate = callback
+end
+-- This could technically be called by clients if they want, it should be more hidden.
+way_cooler_table.handle_termination = function()
```

Comment:
Ok, this needs to be locked down but at least it's not as bad as I expected. Make sure you keep a list of what needs to be locked down as we move forward, so we don't miss anything.